### ISSUE TITLE
Netty Handler has mandatory dependency to Bouncycastle in OSGi metadata

### ISSUE THREAD

**Issue by fipro78** [netty/netty#13168]
### Expected behavior
I can use Netty Handler in an OSGi based project without a Bouncycastle dependency.

### Actual behavior
I need to add org.bouncycastle:bctls-jdk15on:1.69 to get the dependency resolved. Without Bouncycastle in the runtime, I get the following error:

```
  Unresolved requirement: Import-Package: io.netty.handler.timeout; version="[4.1.0,5.0.0)"
    -> Export-Package: io.netty.handler.timeout; bundle-symbolic-name="io.netty.handler"; bundle-version="4.1.87.Final"; version="4.1.87"; uses:="io.netty.channel"
       io.netty.handler [17]
         Unresolved requirement: Import-Package: sun.security.ssl; resolution:="optional"
         Unresolved requirement: Import-Package: sun.security.x509; resolution:="optional"
         Unresolved requirement: Import-Package: org.eclipse.jetty.npn; version="[1.0.0,2.0.0)"; resolution:="optional"
         Unresolved requirement: Import-Package: org.eclipse.jetty.alpn; version="[1.0.0,2.0.0)"; resolution:="optional"
         Unresolved requirement: Import-Package: io.netty.internal.tcnative; version="[2.0.0,3.0.0)"; resolution:="optional"
         Unresolved requirement: Import-Package: org.bouncycastle.asn1.pkcs; version="[1.69.0,2.0.0)"; resolution:="optional"
         Unresolved requirement: Import-Package: org.bouncycastle.asn1.x500; version="[1.69.0,2.0.0)"; resolution:="optional"
         Unresolved requirement: Import-Package: org.bouncycastle.cert; version="[1.69.0,2.0.0)"; resolution:="optional"
         Unresolved requirement: Import-Package: org.bouncycastle.cert.jcajce; version="[1.69.0,2.0.0)"; resolution:="optional"
         Unresolved requirement: Import-Package: org.bouncycastle.jce.provider; version="[1.69.0,2.0.0)"; resolution:="optional"
         Unresolved requirement: Import-Package: org.bouncycastle.jsse
  Unresolved requirement: Import-Package: io.netty.handler.ssl; version="[4.1.0,5.0.0)"
    -> Export-Package: io.netty.handler.ssl; bundle-symbolic-name="io.netty.handler"; bundle-version="4.1.87.Final"; version="4.1.87"; uses:="io.netty.buffer,io.netty.channel,io.netty.handler.codec,io.netty.util,io.netty.util.concurrent,javax.crypto,javax.net.ssl"
```

Checking the MANIFEST.MF file you can see that `org.bouncycastle.jsse` and `org.bouncycastle.jsse.provider` are missing the version constraint and the optional flag. Unfortunately I have not found the place where you configured the import rules for the other packages to be optional.

### Steps to reproduce
Add io.netty.handler to an OSGi runtime.

### Minimal yet complete reproducer code (or URL to code)
I would need to create a complete project setup. Is this really necessary in this case? 

### Netty version
4.1.87.Final

### JVM version (e.g. `java -version`)
17.0.5

### OS version (e.g. `uname -a`)
Windows 10

**Comment by jchrys** [comment]
I think this issue is related https://github.com/netty/netty/pull/13156.
it will be fixed at next release. (4.1.88.Final)

**Comment by normanmaurer** [comment]
Yes... This will be fixed in 4.1.88.Final
